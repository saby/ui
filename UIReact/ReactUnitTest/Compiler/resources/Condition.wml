<ReactUnitTest.Compiler.RenderPropWithoutAdditionalProps>
    <ws:if data="{{ _options.renderContent }}">
        <div>123</div>
    </ws:if>
</ReactUnitTest.Compiler.RenderPropWithoutAdditionalProps>
