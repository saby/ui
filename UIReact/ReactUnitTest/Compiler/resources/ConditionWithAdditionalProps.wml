<ReactUnitTest.Compiler.RenderPropWithAdditionalProps>
    <ws:if data="{{ _options.renderContent }}">
        <div>123{{ content.testProp }}</div>
    </ws:if>
</ReactUnitTest.Compiler.RenderPropWithAdditionalProps>
