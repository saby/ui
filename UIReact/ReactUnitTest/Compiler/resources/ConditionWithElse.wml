<ReactUnitTest.Compiler.RenderPropWithoutAdditionalProps>
    <ws:if data="{{ _options.renderContent }}">
        <div>123</div>
    </ws:if>
    <ws:else>
        <div>456</div>
    </ws:else>
</ReactUnitTest.Compiler.RenderPropWithoutAdditionalProps>
