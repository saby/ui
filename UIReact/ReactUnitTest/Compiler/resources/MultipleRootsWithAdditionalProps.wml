<ReactUnitTest.Compiler.RenderPropWithAdditionalProps>
    <div>123{{ content.testProp }}</div>
    <div>456{{ content.testProp }}</div>
</ReactUnitTest.Compiler.RenderPropWithAdditionalProps>
