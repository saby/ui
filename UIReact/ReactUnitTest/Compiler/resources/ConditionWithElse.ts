import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!ReactUnitTest/Compiler/resources/ConditionWithElse';

export default class ConditionWithElse extends Control {
    _template: TemplateFunction = template;
}
