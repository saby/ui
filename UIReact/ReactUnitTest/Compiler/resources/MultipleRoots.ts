import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!ReactUnitTest/Compiler/resources/MultipleRoots';

export default class MultipleRoots extends Control {
    _template: TemplateFunction = template;
}
