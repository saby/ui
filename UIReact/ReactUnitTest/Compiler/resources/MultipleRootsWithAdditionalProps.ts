import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!ReactUnitTest/Compiler/resources/MultipleRootsWithAdditionalProps';

export default class MultipleRootsWithAdditionalProps extends Control {
    _template: TemplateFunction = template;
}
