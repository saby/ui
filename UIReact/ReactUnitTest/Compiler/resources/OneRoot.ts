import { Control, TemplateFunction } from 'UI/Base';
import * as template from 'wml!ReactUnitTest/Compiler/resources/OneRoot';

export default class OneRoot extends Control {
    _template: TemplateFunction = template;
}
