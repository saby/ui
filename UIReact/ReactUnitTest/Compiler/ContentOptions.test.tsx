/**
 * @jest-environment jsdom
 */
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import OneRoot from './resources/OneRoot';
import MultipleRoots from './resources/MultipleRoots';
import Condition from './resources/Condition';
import ConditionWithElse from './resources/ConditionWithElse';
import OneRootWithAdditionalProps from './resources/OneRootWithAdditionalProps';
import MultipleRootsWithAdditionalProps from './resources/MultipleRootsWithAdditionalProps';
import ConditionWithAdditionalProps from './resources/ConditionWithAdditionalProps';

describe.skip('вставка контентных опций в чистом реакте', () => {
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('Один корень', () => {
        act(() => {
            render(<OneRoot />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it('Несколько корней', () => {
        act(() => {
            render(<MultipleRoots />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it('Условие в корне', () => {
        act(() => {
            render(<Condition renderContent={true} />, container);
        });

        expect(container).toMatchSnapshot();

        unmountComponentAtNode(container);

        act(() => {
            render(<Condition renderContent={false} />, container);
        });

        expect(container).toMatchSnapshot();
    });

    it('Условие с else в корне', () => {
        act(() => {
            render(<ConditionWithElse renderContent={true} />, container);
        });

        expect(container).toMatchSnapshot();

        unmountComponentAtNode(container);

        act(() => {
            render(<ConditionWithElse renderContent={false} />, container);
        });

        expect(container).toMatchSnapshot();
    });

    describe('передача дополнительных опций в контентную опцию', () => {
        it('Один корень', () => {
            act(() => {
                render(<OneRootWithAdditionalProps />, container);
            });

            expect(container).toMatchSnapshot();
        });

        it('Несколько корней', () => {
            act(() => {
                render(<MultipleRootsWithAdditionalProps />, container);
            });

            expect(container).toMatchSnapshot();
        });

        it('Условие в корне', () => {
            act(() => {
                render(
                    <ConditionWithAdditionalProps renderContent={true} />,
                    container
                );
            });

            expect(container).toMatchSnapshot();

            unmountComponentAtNode(container);

            act(() => {
                render(
                    <ConditionWithAdditionalProps renderContent={false} />,
                    container
                );
            });

            expect(container).toMatchSnapshot();
        });
    });
});
