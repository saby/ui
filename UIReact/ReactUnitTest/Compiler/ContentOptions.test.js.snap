// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`вставка контентных опций в чистом реакте Несколько корней 1`] = `
<div>
  <div>
    <div>
      123
    </div>
    <div>
      456
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте Один корень 1`] = `
<div>
  <div>
    <div>
      123
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте Условие в корне 1`] = `
<div>
  <div>
    <div>
      123
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте Условие в корне 2`] = `
<div>
  <div />
</div>
`;

exports[`вставка контентных опций в чистом реакте Условие с else в корне 1`] = `
<div>
  <div>
    <div>
      123
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте Условие с else в корне 2`] = `
<div>
  <div>
    <div>
      456
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте передача дополнительных опций в контентную опцию Несколько корней 1`] = `
<div>
  <div>
    <div>
      123456
    </div>
    <div>
      456456
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте передача дополнительных опций в контентную опцию Один корень 1`] = `
<div>
  <div>
    <div>
      123456
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте передача дополнительных опций в контентную опцию Условие в корне 1`] = `
<div>
  <div>
    <div>
      123456
    </div>
  </div>
</div>
`;

exports[`вставка контентных опций в чистом реакте передача дополнительных опций в контентную опцию Условие в корне 2`] = `
<div>
  <div />
</div>
`;
